lib/perl/Genome/Model/Tools/Sx/Trim/ByPosition.pm

```diff
@@ -0,0 +1,152 @@
+package Genome::Model::Tools::Sx::Trim::ByPosition;
+
+use strict;
+use warnings;
+
+use Genome;
+
+use Genome::Model::Tools::Sx::Split::ByNs;
+use Params::Validate (qw/ :types validate_pos /);
+use Regexp::Common;
+use Try::Tiny;
+
+class Genome::Model::Tools::Sx::Trim::ByPosition {
+    is => 'Genome::Model::Tools::Sx::Base',
+    has => {
+        positions_path => {
+            is => 'File',
+            doc => 'Path to the file of positions to trim.',
+        },    
+    },
+    has_transient_optional => {
+        trim_positions => { is => 'HASH', },
+    },
+};
+
+sub help_brief { "Trim sequences by positions" }
+sub help_detail {
+    <<HELP;
+    The positions file is formatted with one sequence with multiple positions to trim per line. If no positions are given for a sequence, the sequence will be removed. A space follows the sequence name then the positions. Start and stop separated by a dash. Separate multiple trim positions by commas.
+
+    Examples:
+
+    # Trim (remove) entire SEQ1
+    SEQ1
+
+    # Trim bases 1000 to 2000 from SEQ2
+    SEQ2 1000-2000
+
+    # Trim several positions from SEQ3
+    SEQ3 100-200,400-500,600-800
+
+HELP
+}
+
+sub execute {
+    my $self = shift;
+
+    my $trim_positions = $self->load_positions( $self->positions_path );
+    $self->trim_positions($trim_positions);
+
+    $self->_init;
+
+    my ($seqs, @seqs_to_write);
+    while ( $seqs = $self->_reader->read ) {
+        for my $seq ( @$seqs ) {
+            $self->trim_sequence($seq);
+            push @seqs_to_write, $seq if length($seq->{seq});
+        }
+        next if not @seqs_to_write;
+        $self->_writer->write(\@seqs_to_write);
+        @seqs_to_write = ();
+    }
+
+    return 1;
+}
+
+sub load_positions {
+    my ($class, $positions_path) = validate_pos(@_, {isa => __PACKAGE__}, {type => SCALAR});
+
+    my $fh = Genome::Sys->open_file_for_reading($positions_path);
+    my %trim_positions;
+    while ( my $line = $fh->getline ) {
+        chomp $line;
+        my ($seq_id, $positions) = split(/\s+/, $line, 2);
+        $class->fatal_message('Duplicate sequence id in trim positions! %s', $seq_id) if $trim_positions{$seq_id};
+
+        if ( not $positions ) {
+            $trim_positions{$seq_id} = [];
+            next;
+        }
+
+        for my $set ( split(/,/, $positions) ) {
+            my ($start, $stop) = split(/\-/, $set);
+            if ( $RE{num}{int}->matches($stop) ) {
+                $class->fatal_message('Invalid positions for %s! %s', $seq_id, $set) if $start == $stop or $start > $stop;
+            } elsif ( $stop ne 'end' ) {
```

Comment:
I'm putting a comment here just so I can see it vanish when the `end` support is dropped :smile: